tests/unit/moduleapi/propagate.tcl

```diff
@@ -580,6 +599,68 @@ tags "modules" {
                     assert_equal [$replica ttl k1] -1
                 }
 
+                test {module notification on set} {
+                    set repl [attach_to_replication_stream]
+
+                    $master SADD s foo
+
+                    wait_for_condition 500 10 {
+                        [$replica SCARD s] eq "1"
+                    } else {
+                        fail "Failed to wait for set to be replicated"
+                    }
+
+                    $master SPOP s 1
+
+                    wait_for_condition 500 10 {
+                        [$replica SCARD s] eq "0"
+                    } else {
+                        fail "Failed to wait for set to be replicated"
+                    }
+
+                    assert_replication_stream $repl {
+                        {multi}
+                        {select *}
+                        {sadd s foo}
+                        {incr notifications}
+                        {exec}
+                        {multi}
+                        {incr notifications}
+                        {del s}
+                        {incr notifications}
+                        {exec}
```

Comment:
this is not gonna change, when we fix the code in t_set.c, right?
still maybe drop here a comment what makes this case special (explicitly stating that it's not gonna change)

Reply:
This is going to change, notice that the `del` comes after `incr notifications` when we fix the key space notification ordering the `del` will come before. I will add a comment.